docs/source/sandbox/packages.rst

```diff
@@ -67,48 +67,111 @@ an experiment. Then to "create" an environment, ReproNim needs to reinstall the
 
-and additional "overlays" in terms of APT repositories
+  - Snaps may prove difficult for tracing because commands to download
+    and build executibles can be embedded into snap packages
 
-- NeuroDebian
+In addition, we should be aware of specific package repositories that will not
+stand on their own but depend upon specific OS distributions or configurations:
+
+- NeuroDebian - a key source for NeuroImaging Debian/Ubuntu packages
 - other PPAs/APT repositories, e.g. for cran
 
-which wouldn't be sufficient on their own.
-They "define" the ENV and do not "tune it" to "activate"
-They also provide "delivery mechanisms"
+Finally, OS package managers (and related repositories and distributions) are
+typically used to install the language-specific package managers described in
+the next section. Therefore, ReproNim "create" will need to install OS packages
+first, followed by language-specific packages. We may need to allow the
+ReproNim environment specification to allow the user to order the package
+installation across multiple package managers to ensure resolution of
+dependencies.
 
 
-Overlay distributions
-~~~~~~~~~~~~~~~~~~~~~
+Language-Related Package Managers
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 Python
 
-- pypi - pip (as the "delivery mechanism", might be used within conda)
-- virtualenv - virtualenv (as the environment where execution is happening, custom python, ENV changes)
-- Anaconda - conda (see https://www.continuum.io/downloads), ENV changes
-- Conda-Forge - conda (see https://conda-forge.github.io/), ENV changes
+- pip 
+
+  - PyPi Package Index: https://pypi.python.org/pypi
+
+- conda
+
+  - Anaconda Science Platform https://www.continuum.io/downloads
+  - Conda-Forge https://conda-forge.github.io/
 
 Others
 
-- ... - npm (for node.js apps)
+- npm - node.js
+- cpan - Perl
+- CRAN - R
+- brew, linuxbrew, gems - Ruby
 
-Data and generic
+Data Package Managers
+~~~~~~~~~~~~~~~~~~~~~
 
 - DataLad
+
+Environment Configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Pretty much in every "computational environment", environment variables are of
+paramount importance since they instrument invocation and possibly pointers to
+where components would be located when executed. "Overlay" (Non-OS) packages
+rely on adjusting (at least) `PATH` env variable so that components they
+install, possibly overlaying OS-wide installation components, take precedence.
+
+- virtualenv 
+
+  - Impacts the configuration of python environment (where execution is
+    happening, custom python, ENV changes)
+
+- modules
+
+  - http://modules.sourceforge.net
+  - Commonly used on HPC, which is the way to "extend" a POSIX distribution.
+  - We might want to be aware of it (i.e., being able to detect etc), since it
+    could provide at least versioning information which is conventionally
+    specified for every installed "module". It might come handy during `trace`
+    operation.
+
+Others
+
+- ansible
+- chef
+- puppet
+- salt
+- fabric
+
```

Comment:
Those are provisioners ... Not overlay environments like modules and virtualenv are